src/rdkafka_mock_cgrp.c

```diff
@@ -1402,20 +1414,75 @@ rd_kafka_mock_cgrp_consumer_member_find_by_instance_id(
     rd_kafka_mock_cgrp_consumer_member_t *member,
     rd_kafkap_str_t *SubscribedTopicNames,
-    int32_t SubscribedTopicNamesCnt) {
+    int32_t SubscribedTopicNamesCnt,
+    const rd_kafkap_str_t *SubscribedTopicRegex) {
         rd_bool_t changed = rd_false;
-        if (!SubscribedTopicNames)
-                /* When client is sending NULL,
-                 * its subscription didn't change */
+        rd_list_t *new_subscription;
+        int32_t i;
+
+        validate_subscription(SubscribedTopicNames, SubscribedTopicNamesCnt,
+                              SubscribedTopicRegex);
+
+        if (!SubscribedTopicNames &&
+            RD_KAFKAP_STR_IS_NULL(SubscribedTopicRegex) &&
+            !member->subscribed_topic_regex) {
+                /* When client is sending NULL for SubscribedTopicNames and
+                 * SubscribedTopicRegex, its subscription didn't change. If we
+                 * already had a regex, we need to compute the regex again. */
                 return changed;
+        }
 
-        int32_t i;
-        rd_list_t *new_subscription = rd_list_new(
-            SubscribedTopicNamesCnt > 0 ? SubscribedTopicNamesCnt : 1, rd_free);
-        for (i = 0; i < SubscribedTopicNamesCnt; i++) {
-                rd_list_add(new_subscription,
-                            RD_KAFKAP_STR_DUP(&SubscribedTopicNames[i]));
+        if (!RD_KAFKAP_STR_IS_NULL(SubscribedTopicRegex)) {
+                if (member->subscribed_topic_regex) {
+                        rd_kafkap_str_destroy(member->subscribed_topic_regex);
+                        member->subscribed_topic_regex = NULL;
+                }
+                if (RD_KAFKAP_STR_LEN(SubscribedTopicRegex) > 0)
+                        member->subscribed_topic_regex =
+                            rd_kafkap_str_copy(SubscribedTopicRegex);
+        }
+
+        if (SubscribedTopicNames) {
+                if (member->subscribed_topic_names)
+                        rd_list_destroy(member->subscribed_topic_names);
+                member->subscribed_topic_names =
+                    rd_list_new(SubscribedTopicNamesCnt, rd_free);
+                for (i = 0; i < SubscribedTopicNamesCnt; i++) {
+                        rd_list_add(member->subscribed_topic_names,
+                                    rd_strdup(SubscribedTopicNames[i].str));
+                }
+        }
+
+        new_subscription =
+            rd_list_new(member->subscribed_topic_names &&
+                                rd_list_cnt(member->subscribed_topic_names) > 0
+                            ? rd_list_cnt(member->subscribed_topic_names)
+                            : 1,
+                        rd_free);
+
+        if (member->subscribed_topic_names) {
+                rd_list_copy_to(new_subscription,
+                                member->subscribed_topic_names,
+                                rd_list_string_copy, NULL);
+        }
+
+        if (member->subscribed_topic_regex) {
```

Comment:
Given NULL is stored when an empty regex is passed this condition is needed otherwise it would have been 
```suggestion
        if (member->subscribed_topic_regex[0]) {
```